shared/src/main/scala/mlscript/ucs/Desugarer.scala

```diff
@@ -375,7 +375,8 @@ class Desugarer extends TypeDefs { self: Typer =>
         case L(IfBlock(lines)) =>
           lines.foreach { desugarMatchBranch(scrutinee, _, partialPattern, collectedConditions) }
         // This case is rare. Let's put it aside.
-        case L(IfLet(_, _, _, _)) => ???
+        case L(IfLet(_, _, _, _)) =>
+          lastWords("please add this rare case to test files")
```

Comment:
Better leave `???` here. `lastWords` and `die` are for things that _cannot_ happen unless there's a bug in the compiler or in your reasoning.